src/util.js

```diff
@@ -24,6 +24,20 @@ var Util = {
         total: 0,
         message: null
     },
+    
+    countOccurences: function(contentString, searchString) {
+        var count = 0;
+
+        var searchRegex = new RegExp(searchString, "g");
```

Comment:
Good catch.  I'm going to use an index base approach as suggest.  Having the indices will come in handy when fixing ```replaceAll``` so that it doesn't replace strings inside the widget tokens.